frontend/tools/animation-controls.js

```diff
@@ -0,0 +1,177 @@
+import * as components from './basic-components.js';
+
+const COMPONENT_CLASSNAME = 'animation-controls';
+
+let currTime = 0;
+let maxTimeInclusive = 1;
+let showCallback;
+
+export const playing = new components.IconToggle('play/pause', false);
+export const repeating = new components.IconToggle('repeat animation on finish', false);
+export const backward = new components.IconToggle('backward play', false);
+
+const el = {};
+
+export function setup() {
```

Comment:
What a job! Your code to handle the animation is way much cleaner that was mine.